File: src/javascript/Factorial.js
/**
 * 
 * @param {number} n 
 */
function factorial(n) {
    let result = 1;
    for (let index = n; index > 1; index--)
        result = result * index;

    return result;
}

function main(){
    console.log('Fatorial => ', factorial(3));
}

main();

File: src/java/Factorial.java
public class Factorial {

  public static void main(String[] args) {
    System.out.println("Fatorial de 7 : " + Factorial(7));
  }

  public static int Factorial(int x) {
    int total = 1;
    for (int i = 2; i < x + 1; i++) {
      total = total * i;
    }
    return total;
  }
}
